src/helpers.js

```diff
@@ -24,37 +24,75 @@ const promisify = require('@google-cloud/common').util.promisify;
 const gax = require('google-gax');
 
 /*!
- * Find a given image and fire a callback with the appropriate image structure.
+ * Convert non-object request forms into a correctly-formatted object.
  *
- * @param {object} image An object representing what is known about the
- *   image.
+ * @param {object|string|Buffer} request An object representing an
+ *   AnnotateImageRequest. May also be a string representing the path
+ *   (filename or URL) to the image, or a buffer representing the image itself.
+ *
+ * @returns An object representing an AnnotateImageRequest.
+ */
+let _requestToObject = request => {
+  if (is.string(request) === true) {
+    // Is this a URL or a local file?
+    // Guess based on what the string looks like, and build the full
+    // request object in the correct format.
+    if (request.indexOf('://') === -1 || request.indexOf('file://') === 0) {
+      request = {image: {source: {filename: request}}};
+    } else {
+      request = {image: {source: {imageUri: request}}};
+    }
+  } else if (Buffer.isBuffer(request)) {
+    // Drop the buffer one level lower; it will get dealt with later
+    // in the function. This allows sending <Buffer> and {image: <Buffer>} to
+    // both work identically.
+    request = {image: request};
+  }
+  return request;
+};
+
+/*!
+ * Coerce several nicer iterations of "how to specify an image" to the
+ * full sturcture expected by the Vision API.
+ *
+ * @param {object} request An object representing an AnnotateImageRequest.
+ *   It may include `image.source.filename` or a buffer passed to
+ *   `image.content`, which are coerced into their canonical forms by this
+ *   function.
  * @param {function} callback The callback to run.
  */
-var coerceImage = (image, callback) => {
+let _coerceRequest = (request, callback) => {
+  // At this point, request must be an object with an `image` key; if not,
+  // it is an error. If there is no image, throw an exception.
+  if (!is.object(request) || is.undefined(request.image)) {
+    return callback(new Error('No image present.'));
+  }
+
   // If this is a buffer, read it and send the object
   // that the Vision API expects.
-  if (Buffer.isBuffer(image)) {
-    callback(null, {
-      content: image.toString('base64'),
-    });
-    return;
+  if (Buffer.isBuffer(request.image)) {
+    request.image = {content: request.image.toString('base64')};
   }
 
-  // File exists on disk.
-  if (image.source && image.source.filename) {
-    fs.readFile(image.source.filename, {encoding: 'base64'}, (err, blob) => {
-      if (err) {
-        callback(err);
-        return;
+  // If the file is specified as a filename and exists on disk, read it
+  // and coerce it into the base64 content.
+  if (request.image.source && request.image.source.filename) {
+    fs.readFile(
+      request.image.source.filename,
+      {encoding: null},
```

Comment:
Don't pass the `{encoding: null}` object to `fs.readFile` at all. The default value for `encoding` is already `null`, meaning that by default, the `blob` passed to the callback will be a Buffer, the raw bytes of the file.

Reply:
Changed.